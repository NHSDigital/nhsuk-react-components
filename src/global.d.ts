declare module 'nhsuk-frontend/packages/components/header/header.js';
declare module 'nhsuk-frontend/packages/components/checkboxes/checkboxes.js';
declare module 'nhsuk-frontend/packages/components/character-count/character-count.js';
declare module 'nhsuk-frontend/packages/components/tabs/tabs.js';
