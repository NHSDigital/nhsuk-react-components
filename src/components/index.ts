export * from './content-presentation/index.js';
export * from './form-elements/index.js';
export * from './layout/index.js';
export * from './navigation/index.js';
export * from './typography/index.js';
export * from './utils/index.js';
