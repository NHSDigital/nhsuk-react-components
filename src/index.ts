export * from './components';
export * from './patterns';
