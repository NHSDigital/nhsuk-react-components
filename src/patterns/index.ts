export * from './nav-a-z/index.js';
export * from './review-date/index.js';
