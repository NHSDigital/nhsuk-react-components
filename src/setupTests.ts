import { TextDecoder, TextEncoder } from 'util';
import '@testing-library/jest-dom';
import { outdent } from 'outdent';

/**
 * Polyfill TextEncoder/TextDecoder for ReactDOM
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(globalThis, {
  TextEncoder: {
    value: TextEncoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
  TextDecoder: {
    value: TextDecoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
});

/**
 * Polyfill Element methods for NHS.UK frontend error summary
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(Element.prototype, {
  scrollIntoView: { value: jest.fn() },
});

/**
 * Polyfill `matchMedia()` for NHS.UK frontend tabs
 */
Object.defineProperty(globalThis, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: true,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    addEventListener: jest.fn(),
    removeListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

beforeEach(() => {
  const stylesheet = document.createElement('style');

  stylesheet.innerHTML = outdent`
    :root {
      --nhsuk-breakpoint-mobile: 20rem;
      --nhsuk-breakpoint-tablet: 40.0625rem;
      --nhsuk-breakpoint-desktop: 48.0625rem;
      --nhsuk-breakpoint-large-desktop: 61.875rem;
    }
  `;

  // Add styles for NHS.UK frontend checks
  document.head.appendChild(stylesheet);

  // Flag NHS.UK frontend as supported
  document.body.classList.add('nhsuk-frontend-supported');
});
