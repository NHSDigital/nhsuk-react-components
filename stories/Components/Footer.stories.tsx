import React from 'react';
import { Footer } from '../../src';
import { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Footer> = {
  title: 'Components/Footer',
  component: Footer,
};
export default meta;
type Story = StoryObj<typeof Footer>;

export const Standard: Story = {
  render: () => (
    <>
      <div id="restOfThePage" style={{ height: '60vh' }} />
      <Footer>
        <Footer.List>
          <Footer.ListItem href="https://www.nhs.uk/nhs-sites/">NHS sites</Footer.ListItem>
          <Footer.ListItem href="https://www.nhs.uk/about-us/">About us</Footer.ListItem>
          <Footer.ListItem href="https://www.nhs.uk/contact-us/">Contact us</Footer.ListItem>
          <Footer.ListItem href="https://www.nhs.uk/about-us/sitemap/">Sitemap</Footer.ListItem>
          <Footer.ListItem href="https://www.nhs.uk/our-policies/">Our policies</Footer.ListItem>
        </Footer.List>
        <Footer.Copyright>&copy; Crown copyright</Footer.Copyright>
      </Footer>
    </>
  ),
};
