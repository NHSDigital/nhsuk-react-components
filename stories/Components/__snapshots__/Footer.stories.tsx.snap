// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Footer Standard smoke-test 1`] = `
<div id="restOfThePage"
     style="height: 60vh;"
>
</div>
<footer>
  <div class="nhsuk-footer">
    <div class="nhsuk-width-container">
      <h2 class="nhsuk-u-visually-hidden">
        Support links
      </h2>
      <ul class="nhsuk-footer__list">
        <li class="nhsuk-footer__list-item">
          <a class="nhsuk-footer__list-item-link"
             href="https://www.nhs.uk/nhs-sites/"
          >
            NHS sites
          </a>
        </li>
        <li class="nhsuk-footer__list-item">
          <a class="nhsuk-footer__list-item-link"
             href="https://www.nhs.uk/about-us/"
          >
            About us
          </a>
        </li>
        <li class="nhsuk-footer__list-item">
          <a class="nhsuk-footer__list-item-link"
             href="https://www.nhs.uk/contact-us/"
          >
            Contact us
          </a>
        </li>
        <li class="nhsuk-footer__list-item">
          <a class="nhsuk-footer__list-item-link"
             href="https://www.nhs.uk/about-us/sitemap/"
          >
            Sitemap
          </a>
        </li>
        <li class="nhsuk-footer__list-item">
          <a class="nhsuk-footer__list-item-link"
             href="https://www.nhs.uk/our-policies/"
          >
            Our policies
          </a>
        </li>
      </ul>
      <p class="nhsuk-footer__copyright">
        Â© Crown copyright
      </p>
    </div>
  </div>
</footer>
`;
